tests/test_pdstemplates.F90

```diff
@@ -45,22 +44,162 @@ program test_pdstemplates
   print *, 'testing getpdstemplate() with template -1 (nonexistent)...'  
   call getpdstemplate(-1, nummap, map, needext, iret)
   if (iret .eq. 0) stop 11
-  if (pdtlen .ne. nummap) stop 12
+  if (nummap .ne. 0) stop 12
   if (needext) stop 13
 
   print *, 'testing extpdstemplate()...'
+  list = 0
+  map = 0
+  print *, 'Index = -1' ! -- returns without doing anything'
+  call extpdstemplate(-1, list, nummap, map)
+  print *, 'Template without map extension (needext = false)'
+  ! -- returns without doing anything
   call extpdstemplate(0, list, nummap, map)
-  if (nummap .ne. 0) stop 14
+  print *, 'Testing the extendable tempaltes'
+  ! ---- still need to test output i.e. nummap and map ----
+  ! Template number 3
+  call getpdstemplate(3, nummap, list, needext, iret)
+  if (iret .ne. 0 .or. nummap .ne. 31 .or. .not. needext) stop 14
+  call extpdstemplate(3, list, nummap, map)
```

Comment:
This is good but you also need to test that the values returned in map and list are correct.

I suggest you add a bunch of variables which contain the lists you expect to get. Then check the lists you get against them, value by value.

Reply:
I believe that some of the code won't run in extpdstemplate because in some cases it needs a value gotten from the map returned by getpdstemplate to be greater than 1 but the values checked seem to be 1 and so no extension takes place. I'm not sure if the values in the map aren't right or if it's the index being checked but some of the templates with extensions aren't being extended.

Reply:
OK, let's get these tests merged and then I'll take a look...